code/modules/mob/living/living.dm

```diff
@@ -1130,21 +1130,27 @@
 		return TRUE
 	return FALSE
 
-/mob/living/proc/ExtinguishMob()
-	if(on_fire)
+/**
+ * Extinguish all fire on the mob
+ *
+ * This removes all fire stacks, fire effects, alerts, and moods
+ * Signals the extinguishing.
+ */
+/mob/living/proc/extinguish_mob()
+	fire_stacks = 0
+	for(var/obj/effect/dummy/lighting_obj/moblight/fire/F in src)
+		qdel(F)
+	clear_alert("fire")
+	SEND_SIGNAL(src, COMSIG_CLEAR_MOOD_EVENT, "on_fire")
```

Comment:
Looks like this signal should also only happen if the mob was on fire.
Might want to early return if `on_fire` is false already. After all, this is the only proc that changes its value.

Reply:
This is to prevent permanent fire alerts, and moods even when not on fire.
The signal was moved to under the on_fire check because we want it to only signal if we changed the on_fire value.

Reply:
Under which situations a fire alert and mood is added for someone who's not on fire?

Reply:
found a better solution in the next push.